tests/xdsl_opt/test_xdsl_opt.py

```diff
@@ -1,15 +1,100 @@
 from xdsl.xdsl_opt_main import xDSLOptMain
 from contextlib import redirect_stdout
 from io import StringIO
+import pytest
+import os
 
 
 def test_opt():
-    filename = 'tests/xdsl_opt/empty_program.xdsl'
-    opt = xDSLOptMain(args=[filename])
+    opt = xDSLOptMain(args=[])
     assert list(opt.available_frontends.keys()) == ['xdsl', 'mlir']
     assert list(opt.available_targets.keys()) == ['xdsl', 'irdl', 'mlir']
     assert list(opt.available_passes.keys()) == []
 
+
+def test_empty_program():
+    filename = 'tests/xdsl_opt/empty_program.xdsl'
+    opt = xDSLOptMain(args=[filename])
+
+    f = StringIO("")
+    with redirect_stdout(f):
+        opt.run()
+
+    expected = open(filename, 'r').read()
+    assert f.getvalue().strip() == expected.strip()
+
+
+def test_not_module_xdsl():
+    filename = 'tests/xdsl_opt/not_module.xdsl'
+    opt = xDSLOptMain(args=[filename])
+
+    with pytest.raises(Exception) as e:
+        opt.run()
+
+    assert e.value.args[
+        0] == "Expected module or program as toplevel operation"
+
+
+def test_not_module_mlir():
+    filename = 'tests/xdsl_opt/not_module.mlir'
+    opt = xDSLOptMain(args=[filename])
+
+    with pytest.raises(Exception) as e:
+        opt.run()
+
+    assert e.value.args[
+        0] == "Expected module or program as toplevel operation"
+
+
+def test_print_to_file():
+    filename_in = 'tests/xdsl_opt/empty_program.xdsl'
+    filename_out = 'tests/xdsl_opt/empty_program_out.xdsl'
+    opt = xDSLOptMain(args=[filename_in, '-o', filename_out])
+
+    inp = open(filename_in, 'r').read()
+    opt.run()
+    expected = open(filename_out, 'r').read()
+
+    assert inp.strip() == expected.strip()
+
+    open(filename_out, 'w').write("")
+
+
+def test_wrong_target():
```

Comment:
We should change the `xdsl_opt_main` to change target depending on the output file name.